src/engine/sidechain/enginerecord.cpp

```diff
@@ -191,7 +199,42 @@ void EngineRecord::process(const CSAMPLE* pBuffer, const int iBufferSize) {
             // An error occurred.
             emit(isRecording(false, true));
         }
-    } else if (recordingStatus == RECORD_ON) {
+    } else if (recordingStatus == RECORD_SPLIT_CONTINUE) {
+        if (fileOpen()) {
+            closeFile();  // Close file and free encoder.
+            if (m_bCueIsEnabled) {
+                closeCueFile();
+            }
+        }
+        updateFromPreferences();  // Update file location from preferences.
+        if (openFile()) {
+            qDebug() << "Splitting to a new file: "<< m_fileName;
+            m_pRecReady->set(RECORD_ON);
```

Comment:
This can be an (unlikely) race condition if the user has stopped the recording in between. 
It can be solved by using connectValueChangeRequest() and write the value finally by setAndConfirm()

This is done for broadcasting as well: 
https://github.com/mixxxdj/mixxx/blob/8d191bf7ce2d62d24c6e0ffac4b2afd7359afa8a/src/engine/sidechain/enginebroadcast.cpp#L60
In addition, status and enable is separated into two COs